pkg/sdk/identifier_helpers_test.go

```diff
@@ -82,3 +83,42 @@ func TestDatabaseObjectIdentifier(t *testing.T) {
 		assert.Equal(t, `"aaa"."bbb"`, identifier.FullyQualifiedName())
 	})
 }
+
+func TestNewSchemaObjectIdentifierWithArgumentsFromFullyQualifiedName(t *testing.T) {
+	testCases := []struct {
+		RawInput string
+		Input    SchemaObjectIdentifierWithArguments
+		Error    string
+	}{
+		{Input: NewSchemaObjectIdentifierWithArguments(`abc`, `def`, `ghi`, DataTypeFloat, DataTypeNumber, DataTypeTimestampTZ)},
+		{Input: NewSchemaObjectIdentifierWithArguments(`abc`, `def`, `ghi`, DataTypeFloat, "VECTOR(INT, 20)")},
+		{Input: NewSchemaObjectIdentifierWithArguments(`abc`, `def`, `ghi`, "VECTOR(INT, 20)", DataTypeFloat)},
+		{Input: NewSchemaObjectIdentifierWithArguments(`abc`, `def`, `ghi`, DataTypeFloat, "VECTOR(INT, 20)", "VECTOR(INT, 10)")},
+		{Input: NewSchemaObjectIdentifierWithArguments(`abc`, `def`, `ghi`, DataTypeTime, "VECTOR(INT, 20)", "VECTOR(FLOAT, 10)", DataTypeFloat)},
+		// TODO(): Won't work, because of the assumption that identifiers are not containing '(' and ')' parentheses
```

Comment:
what about htis TODO?

Reply:
Forgot to add ticket number, for now It's another limitation that we have to define to move forward, but eventually get back to it (imo it's a bigger analysis, discussion, change to think if csv parser is enough for all Snowflake identifiers and their different versions returned by SHOW or DESC commands).

Reply:
SNOW-1571674